### PR TITLE
feat(docs): add structured data

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Optimize the search engine results so the top actionable items come from a more up-to-date source.

- Tested the results [here](https://search.google.com/test/rich-results/result/r%2Ffaq?id=uF9gdUhtE1DOXbOIjnNHUw)
- Ensured that the ld+json was available statically by running `yarn export` and checking the raw source.

Screen Shot 2022-09-14 at 7 57 20 PM

- The FAQ abides by [Google's rules](https://developers.google.com/search/docs/advanced/structured-data/faqpage)

### CODE REVIEW

File: docs/ui/components/FAQSection/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+export function FAQSection({ name, children }: PropsWithChildren<{ name: string }>) {
+  const mainEntity = useMemo(() => {
+    const faqs: { question: string; answer: string }[] = [];
+
+    Children.forEach(children, child => {
+      if (
+        child &&
+        typeof child === 'object' &&
+        'props' in child &&
+        child?.props?.mdxType === 'Collapsible'
```
Comment (ide) [inline docs/ui/components/FAQSection/index.tsx:15]:
```suggestion
        child.props.mdxType === 'Collapsible'
```
or if needed:
```suggestion
        child!.props!.mdxType === 'Collapsible'
```

File: docs/ui/components/FAQSection/index.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      ) {
+        const content = child.props.children;
+        faqs.push({
+          question: child.props.summary,
+          answer: renderToStaticMarkup(content).replace(
+            /\/\.\.\/\.\.\/\.\.\/\.\.\/\.\.\/\.\.\/\.\.\//g,
```
Comment (ide) [inline docs/ui/components/FAQSection/index.tsx:21]: Is there a way to resolve the URL in a way that doesn't couple this component to the path of the page it's used on? (Also this regex can be replaced with a much more readable string literal, but the bigger issue is that this FAQSection component assumes the links are going to be nested a specific level.)

File: docs/ui/components/FAQSection/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    mainEntity,
+  };
+
+  return (
+    <>
+      <Head>
+        <script
+          type="application/ld+json"
+          dangerouslySetInnerHTML={{ __html: JSON.stringify(ldJson) }}
+        />
```
Comment (ide) [inline docs/ui/components/FAQSection/index.tsx:50]: It's not an issue now but may want to think about how multiple application/ld+json items on a page would work, though perhaps Google supports multiple and it's not a problem.

**Review by ide — CHANGES_REQUESTED** [review]

File: docs/ui/components/FAQSection/index.tsx (hunk 1 above)
Comment (Simek) [reply docs/ui/components/FAQSection/index.tsx:15]: This detection also will not work that easy after the MDX v2 lands, can we somehow avoid iterating over the children and detecting type by `mdxType`?

Comment (byCedric) [reply docs/ui/components/FAQSection/index.tsx:15]: Does MDX v2 change the component type? If not, we might be able to just check on that instead?

Comment (Simek) [reply docs/ui/components/FAQSection/index.tsx:15]: This information is no longer present in MDX v2, you can still access similar information (component name) by getting the `prototype.name` of one of nested fields, but it is definitely not an expected usage.

I one of the places where we were using that field I have opted out validating component name and just fully relay on the Prism class name:
* https://github.com/expo/expo/pull/18947/files#diff-40c0f5bb26ef7cd3f8b52c0b9c1b69a03a9d6bcb32e2b2daa50cb22df04555abL90

File: docs/ui/components/FAQSection/index.tsx (hunk 3 above)
Comment (EvanBacon) [reply docs/ui/components/FAQSection/index.tsx:50]: Google supports multiple.

**Pull request closed without merging**